src/ResourceManagement/MachineLearning/Microsoft.Azure.Management.MachineLearning/Management/WebServiceManagementClient.cs

```diff
@@ -90,6 +90,27 @@ public WebService CreateWebServiceObject(string webServiceDefintionFilePath, str
         /// <param name="overwrite">Optional: If a web service with the same name exists, just go ahead and override it if true. Otherwise, throw an exception.</param>
         public void DeployWebService(WebService webService, bool overwrite = false)
         {
+            if (string.IsNullOrEmpty(webService.CommitmentPlan))
+            {
+                throw new InvalidOperationException("The web service you're trying to deploy does not have a CommitmentPlan, you must set one to deploy.");
+            }
+
+            if (string.IsNullOrEmpty(webService.Location))
+            {
+                throw new InvalidOperationException("The web service you're trying to deploy does not have a Location, you must set one to deploy.");
+            }
+
+            if (string.IsNullOrEmpty(webService.ResourceGroupName))
+            {
+                throw new InvalidOperationException("The web service you're trying to deploy does not have a ResourceGroupName, you must set one to deploy.");
+            }
+
+            if (string.IsNullOrEmpty(webService.StorageAccount.Name) ||
+                string.IsNullOrEmpty(webService.StorageAccount.Key))
+            {
+                throw new InvalidOperationException("The web service you're trying to deploy does not have a valid StorageAccount name/key, you must set one to deploy.");
+            }
```

Comment:
You could put above validation into a Validation method of the WebService class and invoke it here.